packages/flame/lib/src/components/core/component.dart

```diff
@@ -806,21 +824,28 @@ class Component {
     onMount();
     _setMountedBit();
     _lifecycleManager?.finishMounting();
-    if (!existingChild) {
-      _parent!.children.add(this);
-    }
+    _parent!.children.add(this);
+    _reAddChildren();
+    _lifecycleManager?.processQueues();
+    _parent!.onChildrenChanged(this, ChildrenChangeType.added);
+  }
+
+  void _reAddChildren() {
     if (_children != null) {
-      _children!.forEach(
-        (child) => child._mount(parent: this, existingChild: true),
-      );
+      final children = List<Component>.from(_children!, growable: false);
```

Comment:
I wonder if this could be possible to do without creating a new list, or possibly have a temporary list variable outside of the method, or maybe there are no use-cases when this method will be called a lot?

Reply:
I converted this to use a static temporary list.

Reply:
How about changing the name to `componentTreeRoot`? That way there is no "fcs" prefix, and no chance for name clash.